ydb/core/tx/tiering/tier/object.cpp

```diff
@@ -1,62 +1,81 @@
 #include "object.h"
-#include "behaviour.h"
-
-#include <ydb/core/tx/tiering/tier/checker.h>
-
-#include <ydb/services/metadata/manager/ydb_value_operator.h>
-#include <ydb/services/metadata/secret/fetcher.h>
 
 #include <library/cpp/json/writer/json_value.h>
 #include <library/cpp/protobuf/json/proto2json.h>
+#include <library/cpp/uri/uri.h>
 
 namespace NKikimr::NColumnShard::NTiers {
 
-NMetadata::IClassBehaviour::TPtr TTierConfig::GetBehaviour() {
-    static std::shared_ptr<TTierConfigBehaviour> result = std::make_shared<TTierConfigBehaviour>();
-    return result;
+NKikimrSchemeOp::TS3Settings TTierConfig::GetPatchedConfig(
+    const std::shared_ptr<NMetadata::NSecret::ISecretAccessor>& secrets) const {
+    auto config = ProtoConfig;
+    if (secrets) {
+        if (!secrets->GetSecretValue(GetAccessKey(), *config.MutableAccessKey())) {
+            AFL_ERROR(NKikimrServices::TX_TIERING)("error", "cannot_read_access_key")("secret", GetAccessKey().DebugString());
+        }
+        if (!secrets->GetSecretValue(GetSecretKey(), *config.MutableSecretKey())) {
+            AFL_ERROR(NKikimrServices::TX_TIERING)("error", "cannot_read_secret_key")("secret", GetSecretKey().DebugString());
+        }
+    }
+    return config;
 }
 
-NJson::TJsonValue TTierConfig::GetDebugJson() const {
-    NJson::TJsonValue result = NJson::JSON_MAP;
-    result.InsertValue(TDecoder::TierName, TierName);
-    NProtobufJson::Proto2Json(ProtoConfig, result.InsertValue(TDecoder::TierConfig, NJson::JSON_MAP));
-    return result;
-}
+TConclusionStatus TTierConfig::DeserializeFromProto(const NKikimrSchemeOp::TExternalDataSourceDescription& proto) {
+    if (!proto.GetAuth().HasAws()) {
+        return TConclusionStatus::Fail("AWS auth is not defined for storage tier");
+    }
 
-bool TTierConfig::IsSame(const TTierConfig& item) const {
-    return TierName == item.TierName && ProtoConfig.SerializeAsString() == item.ProtoConfig.SerializeAsString();
-}
+    // TODO fix secret owner
+    {
+        auto makeSecretId = [](const TStringBuf& secret) -> TString {
+            return NMetadata::NSecret::TSecretId("root@builtin", secret).SerializeToString();   // ... here
+        };
```

Comment:
What is going on here?

Reply:
Unimplemented part, my major mistake. Now fixed.